File: dfs_recursive.go
package main

// dfsRecursive melakukan pencarian pada flat stack dengan linear recursion
// Relasi rekurensi: T(n) = T(n-1) + 1
// Kompleksitas: O(n)
func dfsRecursive(flatStack []FlatNode, target string, index int) (string, bool) {
	// Operasi dasar: comparison (seperti contoh di modul)
	if flatStack[index].Name == target {
		return flatStack[index].Path, true
	} else {
		// Linear recursion: T(n) = T(n-1) + 1
		// Sama seperti factorial/power di modul
		return dfsRecursive(flatStack, target, index+1)
	}
}

File: dfs_recursive.py
def dfs_recursive(flat_stack, target, index=0):
    if index >= len(flat_stack):
        return "", False
    
    if flat_stack[index].name == target:
        return flat_stack[index].path, True
    else:
        return dfs_recursive(flat_stack, target, index + 1)
